packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -831,4 +841,73 @@ describeCompat("IdCompressor Summaries", "NoCompat", (getTestObjectProvider) =>
 			"Should have the other compressor's session from summary",
 		);
 	});
+
+	it("Container uses short DataStore & DDS IDs", async () => {
+		const container = await createContainer({
+			runtimeOptions: { enableRuntimeIdCompressor: "delayed" },
+		});
+		const defaultDataStore = (await container.getEntryPoint()) as ITestDataObject;
+		// This data store was created in detached container, so it has to be short!
+		assert(
+			defaultDataStore._runtime.id.length <= 2,
+			"short data store ID created in detached container",
+		);
+
+		// Ensure we establish "write" connection - this kicks out loading of ID compressor
+		defaultDataStore._root.set("foo", "bar");
+		await provider.ensureSynchronized();
+
+		// generate one ID to force system to request a range of short IDs.
+		// Without that ID Compressor will not attempt to allocate a block of IDs on an op.
+		defaultDataStore._context.containerRuntime.generateDocumentUniqueId();
```

Comment:
I am not sure I understand this. If this is not called, compressed id will not be used during data store / DDS creation?

Reply:
That's how ID compressor works. I've rephrase comment to hopefully better reflect that:

		// Generate one ID to force ID Compressor to request a range of short IDs on next op.
		// Without that call, ID Compressor will not attempt to allocate a block of IDs on a map op that follows.

Reply:
Changed code based on our discussion.

Reply:
Awesome, looks good!